app/src/main/java/com/example/grocerieswizard/shop/subshop/SubShopAdapter.java

```diff
@@ -8,32 +8,37 @@
 import androidx.recyclerview.widget.RecyclerView;
 
 import com.example.grocerieswizard.databinding.SubRecipeBinding;
-import com.example.grocerieswizard.models.ShoppingItem;
-import com.example.grocerieswizard.models.SubShoppingItem;
+import com.example.grocerieswizard.shop.ShopAdapter;
+import com.example.grocerieswizard.shop.ShoppingItem;
 
 import java.util.ArrayList;
 import java.util.Set;
 
 public class SubShopAdapter extends RecyclerView.Adapter<SubShopAdapter.SubShopViewHolder> {
 
+    private final SubShopOnCheckedChangeListener listener;
     SubRecipeBinding binding;
     private final ArrayList<SubShoppingItem> shoppingItems;
-    private final ShopAdapter parentAdapter;
-
     private SubShoppingItem checkedSubItem;
     private static final String TAG = "SubShopAdapter";
 
 
-    public SubShopAdapter(ShopAdapter parentAdapter) {
-        this.parentAdapter = parentAdapter;
+    public SubShopAdapter(SubShopOnCheckedChangeListener listener) {
         shoppingItems = new ArrayList<>();
+        this.listener = listener;
     }
 
     @NonNull
     @Override
     public SubShopViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
         binding = SubRecipeBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
-        return new SubShopViewHolder(binding);
+        SubShopViewHolder holder = new SubShopViewHolder(binding);
+        binding.subIngredientCheckbox.setOnCheckedChangeListener((buttonView, isChecked) -> {
```

Comment:
Here is a way how to set the listeners in onCreateViewHolder instead of in onBind.